addons/vscode/src/extension.ts

```diff
@@ -81,17 +82,17 @@ export async function getCliPath(extensionPath?: string): Promise<string> {
 	state.configPath = configPath;
 
 	const executableExists = (path: string) => {
-		return new Promise(resolve => {
-			try {
-				const spawnRet = spawn(path, ['--help'], {
-					timeout: 1000, /// 1 second
-				});
-				spawnRet.on('error', () => resolve(false));
-				spawnRet.on('exit', (code: number) => resolve(code === 0));
-			} catch {
-				resolve(false);
-			}
-		});
+		const child = spawnSync(path, ['-V'], { timeout: 1000, encoding: 'utf8' });
+		if (child.error) {
+			throw child.error;
```

Comment:
We are raising an error here, that is not instance of `Error`. in practice it is a bad idea.

Reply:
I refactored a little bit. Now it returns a string | null, where null means OK and string represents the error message